pinot-core/src/main/java/org/apache/pinot/core/query/executor/ServerQueryExecutorV1Impl.java

```diff
@@ -259,6 +260,18 @@ private DataTable processQuery(List<IndexSegment> indexSegments, QueryContext qu
       ExecutorService executorService, @Nullable StreamObserver<Server.ServerResponse> responseObserver, long endTimeMs,
       boolean enableStreaming)
       throws Exception {
+
+    // Validate whether column names in the query are valid
```

Comment:
I am a bit hesitant to add additional check to penalize all queries for catching an infrequent error that.

Reply:
We've already done the similar check in `DataSchemaSegmentPruner` class and the as I mentioned `containsAll` method is called for every indexed segment. 

If the query does include invalid column names, those check in `DataSchemaSegmentPruner` would be skipped. 
If the query doesn't include any invalid columns, the effect here is like to add one more pruning check on one extra segment.

I feel that `DataSchemaSegmentPruner` is still needed, in case if there is some inconsistence between the new and old segments and the table doesn't get reloaded.